### PR TITLE
Feature/2314 save playlist and regions in screen processor

### PR DESCRIPTION
**Pull request by sinejespersen**
#### Link to ticket

https://leantime.itkdev.dk/#/tickets/showTicket/2314

#### Description

Related to: https://github.com/os2display/display-admin-client/pull/259
  - Set `phpdoc_to_comment` to `false`in `.php-cs-fixer.dist.php` to avoid breaking psalm ignore
  - Add regions and groups to `ScreenInput.php`
  - Add "cascade: persist remove" to PlaylistScreenRegion
  - Save playlist/regions in `ScreenProcessor.php` (as an alternative to sending multiple requests)
  - Save groups in `ScreenProcessor.php`
  - Update psalm baseline
  - Add regions/playlists and groups to POST screen test
  - `composer update symfony/* --with-dependencies`
 
#### Screenshot of the result

N/A

### CODE REVIEW

**Review by tuj — CHANGES_REQUESTED** [review]

File: src/State/ScreenProcessor.php [hunk 1]
```diff
@@ -44,6 +54,43 @@ protected function fromInput(mixed $object, Operation $operation, array $uriVari
             $screen->setEnableColorSchemeChange($object->enableColorSchemeChange);
         }
 
+        if (isset($object->regionsAndPlaylists) && isset($screen)) {
```
Comment (turegjorup) [inline src/State/ScreenProcessor.php:57]: Would it be possible to have the same write and read API - `regionsAndPlaylists` is different from `regions`used in the read endpoint.

File: src/State/ScreenProcessor.php [hunk 2]
```diff
@@ -44,6 +56,119 @@ protected function fromInput(mixed $object, Operation $operation, array $uriVari
...
+        // Adding relations for playlist/screen/region
+        if (isset($object->regions) && isset($screen)) {
+            foreach ($object->regions as $regionAndPlaylists) {
+                $region = $this->screenLayoutRegionsRepository->findOneBy(['id' => $regionAndPlaylists['regionId']]);
+
+                if (is_null($region)) {
+                    throw new InvalidArgumentException('Unknown region resource');
+                }
+
+                $newPlaylists = array_map(
```
Comment (tuj) [inline src/State/ScreenProcessor.php:68]: $newPlaylistULIDs

File: src/State/ScreenProcessor.php [hunk 3]
```diff
@@ -44,6 +56,119 @@ protected function fromInput(mixed $object, Operation $operation, array $uriVari
...
+                     * @return Ulid
+                     */
+                    fn ($playlistObject): Ulid => Ulid::fromString($playlistObject['id']), $regionAndPlaylists['playlists']);
+
+                $playlistScreens = $this->playlistScreenRegionRepository->findBy([
+                    'screen' => $screen->getId(),
+                    'region' => $regionAndPlaylists['regionId'],
+                ]);
+
+                $existingPlaylists = array_map(function ($playlistObject) {
```
Comment (tuj) [inline src/State/ScreenProcessor.php:81]: $existingPlaylistULIDs

File: src/State/ScreenProcessor.php [hunk 4]
```diff
@@ -44,6 +56,119 @@ protected function fromInput(mixed $object, Operation $operation, array $uriVari
...
+                        /**
+                         * @param array<mixed> $playlistAndWeight
+                         *
+                         * @return bool
+                         */
+                        fn ($playlistAndWeight) => Ulid::fromString($playlistAndWeight['id']) == $playlist->getId());
+
+                    $playlistAndRegionToSave->setPlaylist($playlist);
+                    $playlistAndRegionToSave->setRegion($region);
+                    $playlistAndRegionToSave->setWeight($playlistWeight[0]['weight']);
```
Comment (tuj) [inline src/State/ScreenProcessor.php:122]: can $playlistWeight have a length of 0 ?

File: src/State/ScreenProcessor.php [hunk 5]
```diff
@@ -44,6 +56,119 @@ protected function fromInput(mixed $object, Operation $operation, array $uriVari
...
+                // ... and saves them.
+                foreach ($newPlaylists as $newPlaylist) {
+                    $playlistAndRegionToSave = new PlaylistScreenRegion();
+                    $playlist = $this->playlistRepository->findOneBy(['id' => $newPlaylist]);
+
+                    if (is_null($playlist)) {
+                        throw new InvalidArgumentException('Unknown playlist resource');
+                    }
+
+                    $playlistWeight = array_filter($regionAndPlaylists['playlists'],
```
Comment (tuj) [inline src/State/ScreenProcessor.php:112]: Add a comment about what this is for

**Review by tuj — CHANGES_REQUESTED** [review]

File: src/State/ScreenProcessor.php (hunk 4 above)
Comment (sinejespersen) [reply src/State/ScreenProcessor.php:122]: no. but i can create a length check before setting weight and alternatively set the weight to 0?

**Review by turegjorup — APPROVED** [review]

**Pull request merged**